tutorials/integration.md

```diff
@@ -599,161 +491,121 @@ use a test network such as Kovan, but such a network is too slow for serious tes
 To do that, you run the tests locally:
 
 
-### Manual Tests
+### Manual Tests <a id="manual_tests"></a>
 
 1. Start a local Ethereum simulator, such as [ganache](https://www.trufflesuite.com/ganache).
+   ::: warning
+   Due to a bug somewhere `chainId` and `netId` need to be identical. To run ganache, use
+   this command line:
+   ```
+   net=`date "+%j%H%M%S"`  && ganache-cli --networkId $net --chainId $net -v
+   ```
+   :::
 1. Make sure that the truffle configuration file (either `truffle.js` or `truffle-config.js`)
-  contains the necessary information to connect to that network.
+   contains the necessary information to connect to that network.
 1. Deploy the GSN contracts:
-+
-[source.bash]
-```
-node_modules/.bin/gsn start
-```
+   ```sh
+   node_modules/.bin/gsn start
+   ```
 1. See the contract numbers and relay URL at the bottom of the output
-+
-```
-
-## startGSN: ready. <a id="startgsn:_ready."></a>
-GSN started
-
-  RelayHub: 0xCA3ef05158d0dBD38cC7B49FbBc979d4cB977Ccc
-  StakeManager: 0x316e9B4bBBBC9B585918CD19357fA686df636D22
-  Penalizer: 0xf905b7E384418de51A74758b31be97D9ef12Ab1F
-  TrustedForwarder: 0x7Aa34e87a62378c1998f2E179EA18200faF866E7
-  Paymaster : 0xC264199C89a1C7056731d2a289B1A5C3fD263CbF
-Relay is active, URL = http://127.0.0.1:44703 . Press Ctrl-C to abort
-```
+   ```
+   GSN started
+
+     RelayHub: 0xCA3ef05158d0dBD38cC7B49FbBc979d4cB977Ccc
+     StakeManager: 0x316e9B4bBBBC9B585918CD19357fA686df636D22
+     Penalizer: 0xf905b7E384418de51A74758b31be97D9ef12Ab1F
+     TrustedForwarder: 0x7Aa34e87a62378c1998f2E179EA18200faF866E7
+     Paymaster : 0xC264199C89a1C7056731d2a289B1A5C3fD263CbF
+   Relay is active, URL = http://127.0.0.1:44703 . Press Ctrl-C to abort
+   ```
 1. Make sure you have the latest versions compiled and start the truffle console.
-+
-```
-truffle compile
-truffle console
-```
+   ```
+   truffle compile
+   truffle console
+   ```
 1. Deploy the paymaster as explained above.
-+
-[source.javascript]
-```
-paymaster = await <paymaster contract>.new()
-paymaster.setRelayHub(<Relay hub address from gsn command>)
-```
+   ```javascript
```

Comment:
why deploy and configure paymaster here? if this part is about client, then use the paymaster deployed by  `gsn start` .
You should use this to get the addresses:
```
const { paymasterAddress, forwarderAddress } = 
    GsnTestEnvironment.loadDeployment()
```